### PR TITLE
Add Spring Cloud App Broker

### PR DESCRIPTION
**Pull request by royclarkson**

### CODE REVIEW

File: start-site/src/main/resources/application.yml [hunk 1]
```diff
@@ -1020,6 +1020,21 @@ initializr:
             - rel: guide
               href: https://github.com/spring-cloud-samples/bookstore-service-broker
               description: Using Spring Cloud Open Service Broker
+        - name: App Broker
```
Comment (snicoll) [inline start-site/src/main/resources/application.yml:1023]: While most Spring Cloud entries are unqualified, I find `App Broker` hard to resonate about, I don't know if `Cloud App Broker` would help with that.

File: start-site/src/main/resources/application.yml [hunk 2]
```diff
@@ -1020,6 +1020,21 @@ initializr:
...
+          id: app-broker
```
Comment (snicoll) [inline start-site/src/main/resources/application.yml:1024]: This sounds like a very generic ID to me. Given that it's open api specific, it should probably be qualified.

File: start-site/src/main/resources/application.yml [hunk 3]
```diff
@@ -1020,6 +1020,21 @@ initializr:
...
+          compatibilityRange: "[2.2.0.RELEASE,2.4.0-M1)"
+          description: Framework for building Spring Boot applications that implement the Open Service Broker API and deploy applications as brokered services.
+          groupId: org.springframework.cloud
+          artifactId: spring-cloud-starter-app-broker-cloudfoundry
```
Comment (snicoll) [inline start-site/src/main/resources/application.yml:1028]: Looking at the current description and name, it is not obvious at all this is Cloudfoundy specific (as I am guessing given the name of the starter).

File: start-site/src/main/resources/application.yml (hunk 1 above)
Comment (royclarkson) [reply start-site/src/main/resources/application.yml:1023]: `Cloud App Broker` works for me if you feel like it makes it more clear. Spring Cloud App Broker is an opinionated implementation framework of Spring Cloud Open Service Broker (or the Open Service Broker API).

File: start-site/src/main/resources/application.yml (hunk 2 above)
Comment (royclarkson) [reply start-site/src/main/resources/application.yml:1024]: is `osbapi-app-broker` better?

File: start-site/src/main/resources/application.yml (hunk 3 above)
Comment (royclarkson) [reply start-site/src/main/resources/application.yml:1028]: App Broker offers an abstraction, but it currently only supports Cloud Foundry. I'll add a bit of clarification.

**Comment by snicoll** [comment]
Thanks Roy. I had mentioned this in the review

> I've looked at the suggestion and documentation and I am a bit confused about the use case and the developer experience of starting a new project including this dependency. Can you please provide some more info?

Can you please provide us with that information?

**Comment by royclarkson** [comment]
@snicoll I'm closing this for now, but may revisit again in the future. Thanks for the feedback.

**Pull request closed without merging**